### PR TITLE
Add ambit

### PR DESCRIPTION
**Pull request by loriab**
Ambit is a tensor manipulation library in C++ with Python bindings through pybind11. It is used mainly by quantum chemistry packages.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
  * ~current commit is v0.6 plus a couple functionality commits plus many buildsystem commits. A v0.7 is in preparation (https://github.com/jturney/ambit/pull/58) and can be finalized when everything needed for conda-forge is settled.~ Using newly-tagged upstream v0.7.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
  * no vendoring
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
  * Linux and Mac libs are dynamic. The Windows library is static, as the dynamic library isn't fully functional.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by loriab** [comment]
@conda-forge/help-c-cpp , ready for review!

File: recipes/ambit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        - if not exist %SP_DIR%\\ambit\\pyambit*.pyd exit 1                    # [win]
+        # Verify accessories
+        - test -f $PREFIX/share/cmake/ambit/ambitConfig.cmake                  # [unix]
+        - test -f $PREFIX/share/cmake/ambit/ambitTargets-Python.cmake          # [unix]
+        - if not exist %PREFIX%\\Library\\share\\cmake\\ambit\\ambitConfig.cmake exit 1  # [win]
+        - if not exist %PREFIX%\\Library\\share\\cmake\\ambit\\ambitTargets-Python.cmake exit 1  # [win]
+        # Inspect linkage
+        - ldd -v $SP_DIR/ambit/pyambit*.so                                     # [linux]
+        - otool -L $SP_DIR/ambit/pyambit*.so                                   # [osx]
+        - objdump.exe -p %SP_DIR%\\ambit\\pyambit.*.pyd | findstr /i "dll"     # [win]
```
Comment (carterbox) [inline recipes/ambit/meta.yaml:121]: For my curosity, what additional information are you gaining by inspecting linkages here? Conda-build already reports linkage information from the binaries in your package.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/ambit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  version: {{ version }}
+
+source:
+  - url: https://github.com/jturney/{{ name }}/archive/v{{ version }}.tar.gz
+    patches:
+      -  0001-lapack-private-to-avoid-overlinking.patch
+    sha256: {{ sha256 }}
+  - url: https://github.com/psi4/iomp5md/archive/31b4ccb.tar.gz               # [win]
+    sha256: d9c6ad3976183f54b676335885d8557d67ca7b25ac4115b91b4cc42281d95f7c  # [win]
+    folder: external_src                                                      # [win]
```
[the remark this answers is not in the document]
Comment (loriab) [reply recipes/ambit/meta.yaml:16]: Right, it's grabbing an import lib to allow compiling against the dll in `intel-openmp  # [win]`. More background is [here at the psi4 recipe](https://github.com/conda-forge/staged-recipes/pull/22328/files#diff-36c704d61400c627caebb094c8cacad1e0dc7a1f63b2085a52facba8ee650ccdR58-R69). Alternatives seem to be allowing different OpenMP libs for MKL and the consuming lib and/or placing `export KMP_DUPLICATE_LIB_OK=TRUE` in an activation/deactivation script. I agree the stashed .lib is unsavory, but it does have the advantage of handling the problem at build-time, and it's time-tested.

File: recipes/ambit/meta.yaml (hunk 1 above)
Comment (loriab) [reply recipes/ambit/meta.yaml:121]: Mostly habit and familiarity of output. The `ldd` gives the full paths and transitive linkage and glibc info that the conda output doesn't. The `objdump` line has been helpful when placed in `bld.bat` to provide some openmp clues _before_ the conda-build dep analysis fails or tests fail. For the final working recipe, the linkage info seemed best to live in tests.

In pre-c-f, pre-conda-build-3 incarnations of these recipes using local build resources, I used to fake a sysroot and then carefully check the glibc versions and library locations. So except for the windows use case above, inspecting linkages is mostly paranoia at this point.

File: recipes/ambit/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (loriab) [reply recipes/ambit/meta.yaml:16]: If the import lib is too objectionable, I can disable threading for Windows. Ambit has never been built for Windows before, so no one's depending on that feature. I'd rather not deal with the KMP_DUPLICATE envvar.

**Comment by loriab** [comment]
@conda-forge/teams/help-c-cpp , ready for review!

@carterbox I just removed the windows since no one needs it. The linux-64 and osx-64 I would like others to use soon. It would eventually be good to know how objectionable the import lib is, so I don't re-engage it or add it elsewhere. Thanks!

**Comment by loriab** [comment]
@conda-forge/help-c-cpp , ready for review!

Whoops, I had the wrong incantation to summon the labeling bot last time.

**Pull request merged**